### PR TITLE
[Docs] add contribution and commit guidelines

### PR DESCRIPTION
**Pull request by e-strauss**
[Docs] add contribution and commit guidelines

- Establishes a standardized framework for commit messages and issue referencing to maintain an organized and scannable project history. 
- This provides clear instructions for contributors on how to categorize changes and link them to trackable work items.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+| `[Fix]` | Bug fixes | `[Fix] correct DAG cycle detection for nested pipelines` |
+| `[Perf]` | Performance enhancements (single commit) | `[Perf] vectorize batch scoring in Rust runtime` |
+| `[Test]` | Test-only changes | `[Test] add coverage for batch executor edge cases` |
+| `[Docs]` | Documentation changes | `[Docs] add API reference for PipelineBuilder` |
+| `[_FEATURE_NAME_]` | Part of a multi-commit feature | `[LazyDAG] add topological sort for execution planning` |
+
+### Feature Tags
+
+For larger features built across multiple commits, use a shared feature tag with the feature name. This makes it easy to trace all commits related to a feature:
+
```
Comment (phaniarnab) [inline CONTRIBUTING.md:31]: Shared feature tag and shared github issue.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+When a commit relates to a GitHub issue, reference it at the end of the commit title:
+
+```
+[Fix] correct DAG cycle detection for nested pipelines #42
+[LazyDAG] add topological sort for execution planning #18
+```
+
+Bigger features should be planned as GitHub issues with sub-issues to break the work into trackable pieces. Each sub-issue maps to one or more commits sharing the same feature tag:
+
```
Comment (phaniarnab) [inline CONTRIBUTING.md:50]: Mention somewhere that most commits should have an associated github issue. Only exception is the Minor tag. (A typical workflow is to first create the initial tasks and sub-tasks for a new feature and then start changing the code.)

**Review by phaniarnab — COMMENTED** [review]
Thanks for writing these up @e-strauss. Please address the comments and merge.

**Comment by e-strauss** [comment]
@phaniarnab  I updated the doc accordingly, please have a look

File: CONTRIBUTING.md (hunk 2 above)
Comment (e-strauss) [reply CONTRIBUTING.md:50]: i added a dedicated workflow section

File: CONTRIBUTING.md (hunk 1 above)
Comment (e-strauss) [reply CONTRIBUTING.md:31]: i added the github issue to he example commits, i dont want to mention the github issue tags i nthe text section here, since we explain in it the following section
